src/gcd.js

```diff
@@ -3,12 +3,24 @@
  * Uses the euclidean algorithm: https://www.youtube.com/watch?v=JUzYl1TYMcU
  * e.g. gcd(8, 12) => 4
  */
+
 function gcd(a, b) {
-  while (a !== 0) {
-    let temp = a
-    a = b % a
+  let big = Math.max(a, b)
+  let small = Math.min(a, b)
+  let r = big % small
+  
+  // if the equation returns no remainder, the GCD = smallest number
+  if (r === 0) {
+      return small
+  }
+
+  while (r !== 0) {
+      r = big % small
+      big = small
+      small = r
+      console.log(r, big, small)
```

Comment:
Purely for comparison, the "concise and cheeky" way of doing this might look something like

```js
function gcd(a, b) {
  let [big, small] = [b, a].sort((x, y) => y - x)
  while (small > 0) [big, small] = [small, big % small]
  return big
}
```

This isn't better or faster, but it's pretty interesting and has some cool things to learn in it

Reply:
Super. Knew there would be a more concise (I'm used to "pythonic" from my background - is there a JS expression akin to this??) way to achieve it.